### PR TITLE
remove unsubscribed sub from subs to fix resending again problem

### PR DESCRIPTION
**Pull request by sajjadkarimi**
Signed-off-by: karimi <skarimi68@gmail.com>

### CODE REVIEW

File: src/main/java/com/streamr/client/StreamrClient.java [hunk 1]
```diff
@@ -585,9 +585,9 @@ private void handleSubscribeResponse(SubscribeResponse res) throws SubscriptionN
         }
     }
 
-    private void handleUnsubcribeResponse(UnsubscribeResponse res) throws SubscriptionNotFoundException {
+    private void handleUnsubscribeResponse(UnsubscribeResponse res) throws SubscriptionNotFoundException {
         Subscription sub = subs.get(res.getStreamId(), res.getStreamPartition());
-        sub.setState(Subscription.State.UNSUBSCRIBED);
```
Comment (hpihkala) [inline src/main/java/com/streamr/client/StreamrClient.java:590]: We should probably keep this line! (The application using the library might hold a reference to the `Subscription` instance, as it is returned from the `subscribe` method. So even though we clean the instance up from `subs`, we should update it to reflect the correct state.)

File: src/main/java/com/streamr/client/StreamrClient.java [hunk 2]
```diff
@@ -585,9 +585,9 @@ private void handleSubscribeResponse(SubscribeResponse res) throws SubscriptionN
         }
     }
 
-    private void handleUnsubcribeResponse(UnsubscribeResponse res) throws SubscriptionNotFoundException {
+    private void handleUnsubscribeResponse(UnsubscribeResponse res) throws SubscriptionNotFoundException {
```
Comment (hpihkala) [inline src/main/java/com/streamr/client/StreamrClient.java:588]: Nice catch! 👍

**Review by hpihkala — CHANGES_REQUESTED** [review]
Very cool to get a contribution from you Sajjad, thanks a lot! One minor request, otherwise looks good to me.

File: src/main/java/com/streamr/client/StreamrClient.java (hunk 1 above)
Comment (sajjadkarimi) [reply src/main/java/com/streamr/client/StreamrClient.java:590]: Thank you, I'll revert that line

**Review by hpihkala — APPROVED** [review]
LGTM, thanks!

**Comment by hpihkala** [comment]
@sajjadkarimi we'll submit the updated version to the Maven Central repository, so you can pull it to your app. It takes a while to do so, but it should go through today or tomorrow!

**Comment by hpihkala** [comment]
@sajjadkarimi version `2.0.2` of the Java client should now be available!

**Comment by sajjadkarimi** [comment]
> @sajjadkarimi version `2.0.2` of the Java client should now be available!

Thank you @hpihkala 
Actually I have tried this version to get historical data in small chunks using repeated resending subscription. and still there is 2 issues related to this subject:
1) `NoResend` event occurs after removing subscription from subs. and then `SubscriptionNotFoundException` in `handleResendResponseNoResend`.

2) Why `resend` function doesn't return Subscription same as `subscribe` method?! it would be helpful awaiting until `sub.getState().equals(Subscription.State.UNSUBSCRIBED)` and continuing next cycle (next resending operation for next chunk).

**Pull request merged**